django/thunderstore/social/api/experimental/views/complete_login.py

```diff
@@ -0,0 +1,67 @@
+from django.conf import settings
+from django.http import HttpRequest, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import serializers
+from rest_framework.response import Response
+from rest_framework.status import HTTP_400_BAD_REQUEST
+from rest_framework.views import APIView
+
+from thunderstore.social.providers import DiscordOauthHelper, GitHubOauthHelper
+
+
+class RequestBody(serializers.Serializer):
+    code = serializers.CharField(
+        label="Temporary code received from provider when authentication flow was initiated on client"
+    )
+    redirect_url = serializers.CharField(
+        label="Redirect URL used when the authentication flow was initiated on client"
+    )
+    secret = serializers.CharField(
+        label="Token shared between Thunderstore services to authenticate OAuth requests"
+    )
+
+
+class ResponseBody(serializers.Serializer):
+    session_id = serializers.CharField()
+
+
+class CompleteLoginApiView(APIView):
+    """
+    Complete OAuth login process initiated by a client.
+    """
+
+    permission_classes = []
+
+    @swagger_auto_schema(
+        operation_id="experimental.auth.complete",
+        request_body=RequestBody,
+        responses={200: ResponseBody()},
+    )
+    def post(self, request: HttpRequest, provider: str) -> HttpResponse:
+        secret = request.data.get("secret")
+        code = request.data.get("code")
+        redirect_url = request.data.get("redirect_url")
+
+        if not secret or secret != settings.OAUTH_SHARED_SECRET:
+            return Response("Invalid secret", HTTP_400_BAD_REQUEST)
+
+        if not code or not len(code):
+            return Response("Invalid code parameter", HTTP_400_BAD_REQUEST)
+
+        if not redirect_url or not len(redirect_url):
+            return Response("Invalid redirect_url parameter", HTTP_400_BAD_REQUEST)
+
+        if provider.lower() == "discord":
```

Comment:
Perhaps this could already be turned into a standalone method for getting a helper by provider name, and if it returns `None` then throw the error?

Reply:
Done.